alibi/explainers/shap_wrappers.py

```diff
@@ -730,20 +732,20 @@ def fit(self,  # type: ignore
 
         # check user inputs to provide warnings if input is incorrect
         self._check_inputs(background_data, group_names, groups, weights)
-        if self.create_group_names:
+        if self.create_group_names and groups:
```

Comment:
We don't check if `groups` is `None` beforehand, so type-checker complains that `len(groups)` may fail.